### PR TITLE
Credentials Rotation Implement

### PR DESCRIPTION
**Pull request by melamedbn**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Must have
- [ ] Tests
- [x] Documentation

### CODE REVIEW

File: Packs/CloudIncidentResponse/Playbooks/playbook-Cloud_IAM_User_Access_Investigation.yml [hunk 1]
```diff
@@ -1019,6 +1206,23 @@ inputs:
   required: false
   description: Whether to execute the user remediation flow automatically.
   playbookInputQuery:
+- key: credentialsRemediationType
+  value: {}
```
Comment (altmannyarden) [inline Packs/CloudIncidentResponse/Playbooks/playbook-Cloud_IAM_User_Access_Investigation.yml:1210]: if we dont have a default value, we should add if this input isnt empty for the conditional task whether to run the creds rotation playbook

File: Packs/CloudIncidentResponse/Playbooks/playbook-Cortex_XDR_-_Cloud_Data_Exfiltration_Response.yml [hunk 2]
```diff
@@ -1047,19 +1048,176 @@ tasks:
...
     isoversize: false
     isautoswitchedtoquietmode: false
+  "86":
+    id: "86"
+    taskid: b4d64aed-f0ba-4f76-83a6-67e469740ba7
+    type: playbook
+    task:
+      id: b4d64aed-f0ba-4f76-83a6-67e469740ba7
+      version: -1
+      name: Cloud Credentials Rotation - Generic
```
Comment (altmannyarden) [inline Packs/CloudIncidentResponse/Playbooks/playbook-Cortex_XDR_-_Cloud_Data_Exfiltration_Response.yml:1058]: In the other playbook, you added a conditional task whether to run this subplaybook. why don't we have it here?

File: Packs/CloudIncidentResponse/Playbooks/playbook-Cortex_XDR_-_Cloud_Data_Exfiltration_Response.yml [hunk 3]
```diff
@@ -1082,6 +1240,48 @@ inputs:
   required: false
   description: 'Whether to execute the block remediation automatically. (Default: False)'
   playbookInputQuery:
+- key: credentialsRemediationType
+  value:
+    simple: Reset
```
Comment (altmannyarden) [inline Packs/CloudIncidentResponse/Playbooks/playbook-Cortex_XDR_-_Cloud_Data_Exfiltration_Response.yml:1245]: add this input in the conditional task whether to run the creds rotation or not

File: Packs/CloudIncidentResponse/Playbooks/playbook-Cortex_XDR_-_Cloud_IAM_User_Access_Investigation.yml [hunk 4]
```diff
@@ -759,6 +878,23 @@ inputs:
   required: false
   description: Whether to execute the user remediation flow automatically.
   playbookInputQuery:
+- key: credentialsRemediationType
+  value: {}
```
Comment (altmannyarden) [inline Packs/CloudIncidentResponse/Playbooks/playbook-Cortex_XDR_-_Cloud_IAM_User_Access_Investigation.yml:882]: need to check if this input isnt empty

File: Packs/CloudIncidentResponse/Playbooks/playbook-XCloud_Cryptomining.yml [hunk 5]
```diff
@@ -976,6 +979,193 @@ tasks:
...
+      iscommand: false
+      brand: ""
+    nexttasks:
+      "yes":
+      - "59"
+    separatecontext: false
+    conditions:
+    - label: "yes"
+      condition:
+      - - operator: isEqualString
```
Comment (altmannyarden) [inline Packs/CloudIncidentResponse/Playbooks/playbook-XCloud_Cryptomining.yml:1001]: Probably need to add here the is not empty for the creds type input

File: Packs/CloudIncidentResponse/Playbooks/playbook-XCloud_Cryptomining.yml [hunk 6]
```diff
@@ -476,6 +477,8 @@ tasks:
           root: inputs.alert_id
       extend-context:
```
[the remark this answers is not in the document]
Comment (melamedbn) [reply Packs/CloudIncidentResponse/Playbooks/playbook-XCloud_Cryptomining.yml:478]: No need, we're using the PaloAltoNetworks context key instead.

**Review by altmannyarden — APPROVED** [review]

**Comment by DeanArbel** [comment]
The build was mostly successful but had constant unrelated issues.

**Pull request merged**